src/command/process_update_task.rs

```diff
@@ -162,46 +208,208 @@ impl UpdateProcessor {
+                    .get_city_row(&self.chat.selected.clone().unwrap(), number)
+                    .await?;
+
                 self.return_to_initial().await?;
 
-                self.pattern_response(number).await
+                self.get_weather(city).await
             }
 
             Err(_) => self.not_number_message().await,
         }
     }
 
-    async fn pattern_response(&self, number: usize) -> Result<(), BotError> {
-        let city = self
-            .repo
-            .get_city_row(&self.chat.selected.clone().unwrap(), number)
-            .await?;
+    async fn process_set_city_number(&self) -> Result<(), BotError> {
+        match self.text.parse::<usize>() {
+            Ok(number) => {
+                let city = self
+                    .repo
+                    .get_city_row(&self.chat.selected.clone().unwrap(), number)
+                    .await?;
+
+                self.return_to_initial().await?;
+
+                self.set_default_city(city).await
+            }
+
+            Err(_) => self.not_number_message().await,
+        }
+    }
+
+    async fn not_valid_offset_message(&self) -> Result<Option<ScheduleWeatherTask>, BotError> {
+        self.cancel(Some(
+            "That's not a valid offset, it has to be a number in range [-11, 12].\n
+            If your timezone is UTC + 2 put 2, if you have UTC - 10 put -10, 0 if you have UTC timezone.\n
+            The command was cancelled"
+            .to_string(),
+        ))
+        .await?;
+
+        Ok(None)
+    }
+
+    async fn process_offset(&self) -> Result<Option<ScheduleWeatherTask>, BotError> {
+        match self.text.parse::<i8>() {
+            Ok(number) => {
+                if !(-11..=12).contains(&number) {
+                    return self.not_valid_offset_message().await;
+                }
+
+                let offset = number;
+                // we have hour and minutes well formatted stored in selected see in process_time func.
+                let vec: Vec<&str> = self.chat.selected.as_ref().unwrap().split(':').collect();
+                // This unwraps are safe because we know that is stored in the correct format.
+                let user_hour = vec[0].parse::<i8>().unwrap();
+                let minutes = vec[1].parse::<i8>().unwrap();
+
+                let hour_utc = if user_hour - offset < 0 {
+                    user_hour - offset + 24
+                } else {
+                    user_hour - offset
+                };
+
+                let cron_expression = format!("0 {} {} * * * *", minutes, hour_utc);
+
+                let schedule_weather_task = ScheduleWeatherTask::builder()
+                    .username(self.username.clone())
+                    .chat_id(self.chat.id)
+                    .user_id(self.chat.user_id)
+                    .cron_expression(cron_expression)
+                    // Safe unwrap checked in process_time func.
+                    // We have to ask about city to get this :(, have to be done
```

Comment:
We have to ask the city if we want to make it work for multiple cities, for now it is in default_city_id but this will change.

Reply:
Done.